apps/editing-toolkit/editing-toolkit-plugin/common/index.php

```diff
@@ -258,13 +262,22 @@ function enqueue_override_preview_button_url() {
 		return;
 	}
 
-	wp_enqueue_script(
-		'a8c_override_preview_button_url',
-		plugins_url( 'dist/override-preview-button-url.min.js', __FILE__ ),
-		array(),
-		filemtime( plugin_dir_path( __FILE__ ) . 'dist/override-preview-button-url.min.js' ),
-		true
-	);
+	if ( empty( $response->data ) || empty( $response->data['link'] ) ) {
+		return $response;
+	}
+
+	// logmein is a special URL that logs users into their custom mapped domain and sets up
+	// their log in cookie in a first-party context.
+	$query_params = wp_parse_url( $response->data['link'], PHP_URL_QUERY );
+
+	if ( $query_params ) {
+		$response->data['link'] = $response->data['link'] . '&logmein=direct';
+	} else {
+		$response->data['link'] = $response->data['link'] . '?logmein=direct';
+	}
```

Comment:
The global function `add_query_arg` looks like it would be useful here and simplify this a bit.

Reply:
Thanks I'll make the changes

Reply:
Done https://github.com/Automattic/wp-calypso/pull/66370/commits/f8a1036f884eea15de9c49bf0c9c2bcd9ee122be